src/librustdoc/passes/doc_test_lints.rs

```diff
@@ -75,6 +77,11 @@ crate fn should_have_doc_example(cx: &DocContext<'_>, item: &clean::Item) -> boo
         return false;
     }
     let hir_id = cx.tcx.hir().local_def_id_to_hir_id(item.def_id.expect_local());
+    if cx.tcx.hir().attrs(hir_id).lists(sym::doc).has_word(sym::hidden)
+        || inherits_doc_hidden(cx, hir_id)
```

Comment:
Why not make this check part of `inherits_doc_hidden`? I see in `visit_ast` there's some different behavior depending on if `doc(hidden)` is inherited or specifically for that item - is the distinction necessary?

Reply:
I answered the first question [above](https://github.com/rust-lang/rust/pull/76716#discussion_r575709432). As for the second: I think it's fine as is, otherwise we'd have to either make an extra check or to duplicate the function.